### ISSUE TITLE
metrics VM generation scripts are not multi-architecture aware

### ISSUE THREAD

**Issue by grahamwhaley** [kata-containers/ci#35]
The VM generation scripts (in VMs/metrics - from #31) are not multi-architecture aware, and are hard wired (in the user-env files) to use the x86 apt repos for instance.
I've added a check so they now report and quit for any non x86_64 invocations, but if/when we get to use these on a non-x86 arch, we should fix the scripts (most likely by making arch specific user-env files if we need to modify the package or commands, or by doing a sed substitute on the arch, like we do for the USER presently).

**Comment by Pennyzct** [comment]
Hi~ correct me if I’m wrong☺️, the hard-coded arch info exists in ‘Install go’ and ‘Install docker’, and official binary has already supported arm64. Only need substitute amd64 for arm64. Doing a sed substitute may be enough?

**Comment by jodh-intel** [comment]
Hi @Pennyzct - it's the version of go, docker and ubuntu that need changing. I guess in theory a `sed` would work but ideally the script would set `arch=$(arch)` and then use `$arch` rather than any hard-coded architecture values. We've tried to use that technique across all repos for architecture-specifics.

**Comment by grahamwhaley** [comment]
I can try it out with a `$(arch)` in place. I don't think we can use `arch=$(arch)` directly in the `cloud-init` files, as each line runs in a separate shell afaik, but using `$(arch)` at each point should work I think.

**Comment by jodh-intel** [comment]
Yep. I think we're going to have to explore the "build the cloud-init files" option and use one of the following depending on the format of the architecture we need:

- `arch=$(arch)`
- `arch=$(go env GOARCH)`

Regarding multiple snippets of cloud-init yaml, it would be interesting to know if it can cope with something like:

```
$ (cat file1.yaml;echo ---; cat file2.yaml) > ./cloud-init.yaml
```

**Comment by grahamwhaley** [comment]
That is a multi-content `yaml` file is it @jodh-intel  - the `---`? - I've not seen that in cloud-init files, but it is used in ccloudvm - which combines a cloud-init file with a file of its own into a single file afaik.
I'll have another look to see if you can have multi-part `cloud-init` files, but when I looked before:
- catting files together does not work, as if you have duplicate sections cloud-init just takes the last one seen
- merging cloud-init files is, err, mildly complex: http://cloudinit.readthedocs.io/en/latest/topics/merging.html
- even then, I have a feeling that is only for multiple files, and we only have the option for a single file with the ISO file [nocloud](http://cloudinit.readthedocs.io/en/latest/topics/datasources/nocloud.html) method we are using
- and merging two yaml files is also non-trivial afaict

Right now I see two prudent paths:
- we tackle the multi-arch another day, unless one of the other arch's want/need to test or use these scripts right now (which is fine btw, just shout!)
- The `sed` solution is sounding quite nice at this point...

**Comment by Pennyzct** [comment]
Hi~ @jodh-intel It's kind of inconvenient in Arm, since `$(arch)` displays aarch64 instead of arm64. So it may need further normalization. But `arch=$(go env GOARCH)` will output the right arm64😭.

**Comment by grahamwhaley** [comment]
looks like `go env GOARCH` also returns the good value for `amd64`.
Now, we just need to get GO installed in the VM so we can use that.... - [catch22](https://en.wikipedia.org/wiki/Catch-22_(logic)) - we need GO installed to know which GO to install :-)

**Comment by jodh-intel** [comment]
Hi @grahamwhaley, @Pennyzct - yes, it's rather distressing that `arch(1)` still doesn't provide an option to allow it to provide both values ((`x86_64` + `amd64`), (`aarch64` + `arm64`)) :frowning_face:.

**Comment by Weichen81** [comment]
Hi @jodh-intel, don't feel so distressing : ) 
I think, at the first, we can select a preferred value for our architectures. IMO, I will select
the `arm64` for 64-bit arm architecture.  This is chosen by most  open source projects, e.g., Linux, FreeBSD. And then use `sed` to normalize `$(arch)` , `$(go env GOARCH)`  or any other applications' output to `arm64`. We can select one and then normalize anther, don't try to avoid `sed`, we will escape from [catch-22](https://en.wikipedia.org/wiki/Catch-22_(logic)) ; )

**Comment by jodh-intel** [comment]
@Weichen81 - that sounds like a good idea. In fact, maybe we could create a simple shell script (`kata-arch.sh`?) in https://github.com/kata-containers/tests/tree/master/.ci that basically does that translation.

By putting it in the `tests` repo, it will be accessible to all the other repos and would avoid in some cases having to install `golang` just to find out the architecture :)

wdyt @grahamwhaley, @chavafg?

**Comment by grahamwhaley** [comment]
I think having a single place that does the translation is a good idea - I also wonder if the `go` trick works for ppc64 (/cc @nitkon).
But, for this PR, I'm not sure it works as this is not quite your 'normal repo build' - this is a cloud-init script running inside a VM that is 'under construction' :-)
I am testing the `sed` fix right now - expect I'll post an update rsn.

**Comment by nitkon** [comment]
@grahamwhaley : ```go env GOARCH``` returns ppc64le
